lib/web_ui/lib/painting.dart

```diff
@@ -405,6 +405,20 @@ class ImageFilter {
     return engine.EngineImageFilter.blur(sigmaX: sigmaX, sigmaY: sigmaY, tileMode: tileMode);
   }
 
+  // ignore: avoid_unused_constructor_parameters
+  factory ImageFilter.dilate({ double radiusX = 0.0, double radiusY = 0.0 }) {
+    // TODO(fzyzcjy): implement dilate.
+    throw UnimplementedError(
+        'ImageFilter.dilate not implemented for web platform.');
+  }
+
+  // ignore: avoid_unused_constructor_parameters
+  factory ImageFilter.erode({ double radiusX = 0.0, double radiusY = 0.0 }) {
+    // TODO(fzyzcjy): implement erode.
+    throw UnimplementedError(
+        'ImageFilter.erode not implemented for web platform.');
```

Comment:
Please file or link to existing github issues for this.

The CanvasKit implementations shouldn't be too hard. The HTML backend might be harder.

Reply:
Yes, please look into implementations here if at all possible. I know it's a fairly big additional body of code to sift through, but we should make it a priority not to have the DOM/canvaskit implementations fall behind as we add new features.

CC'ing @yjbanov

Reply:
HTML backend seems nontrivial. I see some people use svg to do so, but I do not think it is valid for this context. Not find any solutions with CSS (unlike blur - which has CSS implementation).

Reply:
In addition, I guess the `compose` is also not implemented:

  // TODO(ferhat): add implementation and remove the "ignore".
  // ignore: avoid_unused_constructor_parameters
  ImageFilter.compose({required ImageFilter outer, required ImageFilter inner}) {
    throw UnimplementedError(
        'ImageFilter.compose not implemented for web platform.');
  
}

Reply:
Github link added.